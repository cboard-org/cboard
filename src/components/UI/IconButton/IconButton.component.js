import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Tooltip from 'material-ui/Tooltip';
import MUIIconButton from 'material-ui/IconButton';

const propTypes = {
  children: PropTypes.node.isRequired,
  label: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  component: PropTypes.func,
  to: PropTypes.string,
  onClick: PropTypes.func
};

function IconButton({ children, component, to, label, disabled, onClick }) {
  const child = React.Children.only(children);
  const iconButtonProps = {
    'aria-label': label,
    disabled,
    component,
    to,
    onClick,
    color: 'inherit'
  };
  return (
    <Fragment>
      {disabled ? (
        <MUIIconButton
          {...iconButtonProps}
          style={{
            color: 'rgba(255, 255, 255, 0.26)'
          }}
        >
          {child}
        </MUIIconButton>
      ) : (
        <Tooltip title={label} placement="bottom">
          <MUIIconButton {...iconButtonProps}>{child}</MUIIconButton>
        </Tooltip>
      )}
    </Fragment>
  );
}

IconButton.propTypes = propTypes;

export default IconButton;
