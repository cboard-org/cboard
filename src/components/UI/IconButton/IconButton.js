import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Tooltip from '@material-ui/core/Tooltip';
import MUIIconButton from '@material-ui/core/IconButton';

const propTypes = {
  /**
   *
   */
  children: PropTypes.node.isRequired,
  /**
   *
   */
  component: PropTypes.func,
  /**
   *
   */
  disabled: PropTypes.bool,
  /**
   *
   */
  label: PropTypes.string.isRequired,
  /**
   *
   */
  onClick: PropTypes.func,
  /**
   *
   */
  to: PropTypes.string
};

function IconButton({ children, component, to, label, disabled, onClick }) {
  const child = React.Children.only(children);

  const iconButtonProps = {
    component,
    disabled,
    onClick,
    to,
    'aria-label': label,
    color: 'inherit'
  };

  const disableIconButtondStyle = {
    color: 'rgba(255, 255, 255, 0.26)'
  };

  return (
    <Fragment>
      {disabled ? (
        <MUIIconButton {...iconButtonProps} style={disableIconButtondStyle}>
          {child}
        </MUIIconButton>
      ) : (
        <Tooltip placement="bottom" title={label}>
          <MUIIconButton {...iconButtonProps}>{child}</MUIIconButton>
        </Tooltip>
      )}
    </Fragment>
  );
}

IconButton.propTypes = propTypes;

export default IconButton;
