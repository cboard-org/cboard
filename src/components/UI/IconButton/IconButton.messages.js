import { defineMessages } from 'react-intl';

export default defineMessages({
  back: {
    id: 'cboard.components.BackButton.back',
    defaultMessage: 'Go back'
  }
});
