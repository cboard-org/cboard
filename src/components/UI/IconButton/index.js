export { default } from './IconButton.component';
