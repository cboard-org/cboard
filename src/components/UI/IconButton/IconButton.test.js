import React from 'react';
import { shallowMatchSnapshot } from '../../../common/test_utils';

import IconButton from './IconButton';

describe('IconButton tests', () => {
  test('default renderer', () => {
    shallowMatchSnapshot(
      <IconButton label="Go back">
        <div>icon</div>
      </IconButton>
    );
  });
});
