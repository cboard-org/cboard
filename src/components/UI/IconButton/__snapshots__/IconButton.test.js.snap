// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton tests default renderer 1`] = `
<React.Fragment>
  <WithStyles(Tooltip)
    placement="bottom"
    title="Go back"
  >
    <WithStyles(IconButton)
      aria-label="Go back"
      color="inherit"
    >
      <div>
        icon
      </div>
    </WithStyles(IconButton)>
  </WithStyles(Tooltip)>
</React.Fragment>
`;
